src/mpid/ch4/netmod/ofi/ofi_pre.h

```diff
@@ -129,17 +129,27 @@ typedef struct {
         uint64_t lmt_cntr;
         MPIDI_OFI_lmt_unpack_t unpack;
     } lmt_u;
-    void *pack_buffer;
     MPIR_Request *rreq_ptr;
     void *am_hdr;
     uint16_t am_hdr_sz;
-    uint8_t pad[6];
```

Comment:
Do you know why this padding was needed before and why it's not necessary anymore?

Reply:
From the look of it, it is trying to pad `am_hdr_sz` into 8-byte. But since the `msg_hdr` field itself is at 8-byte alignment, I don't see we need manually pad it.

Reply:
That's fine, but `msg_hdr` isn't next to `am_hdr_sz` anymore.

Reply:
We only need to worry about the alignment of `msg_hdr` and `am_hdr_buf`. It is already at 8-byte. Let me check, maybe we want to force it at 16-byte or `MAX_ALIGNEMENT`. Either way, this explicit `pad[6]` is obscure and fragile. We probably want to add an `MPL_ATTR_ALIGNED`.